jadx-gui/src/main/java/jadx/gui/ui/SearchDialog.java

```diff
@@ -246,8 +248,6 @@ public void keyReleased(KeyEvent e) {
 		});
 
 		setTitle(NLS.str("menu.text_search"));
-		pack();
-		setSize(800, 500);
```

Comment:
Hm, if you don't call `pack()` or `setSize` first run will show zero size window.
So I suggest to check if window pos not applied and call `pack()`: 
```java
if (!mainWindow.getSettings().loadWindowPos(this)) {
		pack();
}
```
Also, window pos is ignored for some cases: like moving window outside of screen bounds (check `JadxSettings#loadWindowPos()` code and PR #521).

Reply:
You are correct, I did not consider the case when Jadx is started the first time. I will make some more tests. 

Unfortunately your proposed changes doesn't fix all problems. I assume that there is a bug in the calculation of WrapLayout. I will check that tomorrow.

Reply:
It took a little bit longer to check the WrapLayout. I updated the code so the dialog size is now set if the dialog position can not be restored from settings.